src/mrsimulator/signal_processing/baseline.py

```diff
@@ -124,3 +124,50 @@ def operate(self, data):
         """
         data += self.offset
         return data
+
+
+class Polynomial(Baseline):
+    r"""Add a baseline polynomial to all dependent variables (y) in the CSDM object.
+
+    The baseline function is
+
+    .. math::
+            f(x) = \sum_{i=0}^n x_i \times x^i,
+
+    where :math:`x` is the CSDM dimension at index ``dim_index``.
+
```

Comment:
@mVenetos97 I was reading the doc string in the html docs and it's a bit confusing. Can you change the coefficients from `x_i` to `c_i` and add a sentence at the end describing what `c_i` is? For example, `where c_i are the coefficients corresponding to x^i`.